server/src/main/java/org/apache/uniffle/server/ShuffleServerMetrics.java

```diff
@@ -79,6 +79,8 @@ public class ShuffleServerMetrics {
   private static final String USED_BUFFER_SIZE = "used_buffer_size";
   private static final String READ_USED_BUFFER_SIZE = "read_used_buffer_size";
   private static final String USED_DIRECT_MEMORY_SIZE = "used_direct_memory_size";
+  private static final String ALLOCATED_DIRECT_MEMORY_SIZE = "allocated_direct_memory_size";
+  private static final String PINNED_DIRECT_MEMORY_SIZE = "pinned_direct_memory_size";
```

Comment:
What's the meaning of `pinned`?

Reply:
Refer to https://github.com/netty/netty/pull/1166. 

To be brief, it means the direct memory actually used by `PooledByteBufAllocator`,  which cannot be used anymore. If we use the old metric `usedDirectMemory`, we cannot tell whether the diect memory is cached by `PooledByteBufAllocator` or actually used.

Reply:
Could you give it a better name?